From 73f0f52145f25a38afba669ba5091a63e8182fff Mon Sep 17 00:00:00 2001
From: Gyorgy Sarvari <skandigraun@gmail.com>
Date: Mon, 4 Nov 2024 16:52:08 +0100
Subject: [PATCH] fix compiling with python 3.13

Walnascar comes with Python 3.13, which has the pipes module removed.
Due to this, the compilation of the latest Firefox fails on this
Yocto release. This patch changes the use of `pipes.quote()` function to
its recommended substitute, `shlex.quote()` module.

Upstream-Status: Inappropriate [OE-specific]
---
 python/mozbuild/mozbuild/action/node.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/python/mozbuild/mozbuild/action/node.py b/python/mozbuild/mozbuild/action/node.py
index fca0745..d86e58a 100644
--- a/python/mozbuild/mozbuild/action/node.py
+++ b/python/mozbuild/mozbuild/action/node.py
@@ -2,7 +2,7 @@
 # License, v. 2.0. If a copy of the MPL was not distributed with this
 # file, You can obtain one at http://mozilla.org/MPL/2.0/.
 
-import pipes
+import shlex
 import subprocess
 import sys
 
@@ -47,7 +47,7 @@ def execute_node_cmd(node_cmd_list):
     """
 
     try:
-        printable_cmd = " ".join(pipes.quote(arg) for arg in node_cmd_list)
+        printable_cmd = " ".join(shlex.quote(arg) for arg in node_cmd_list)
         print('Executing "{}"'.format(printable_cmd), file=sys.stderr)
         sys.stderr.flush()
 
-- 
2.47.0

